mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -981,42 +1133,12 @@ private class LogisticAggregator(
       }
       val localGradientSumArray = gradientSumArray
```

Comment:
Since you are passing `localGradientSumArray` into a function, you don't need to copy it anymore. Just pass `gradientSumArray` into the function.